esbuild.config.js

```diff
@@ -0,0 +1,54 @@
+const baseConfig = {
+  bundle: true,
+  platform: "node",
+  target: "node20",
+  sourcemap: true,
+  treeShaking: true,
+  external: [],
+  splitting: false,
+  logLevel: "info",
+};
+
+const targets = [
+  {
+    ...baseConfig,
+    entryPoints: ["src/main.entrypoint.ts"],
+    outfile: "dist/main.entrypoint.js",
+    format: "esm",
+    banner: {
+      js: 'import { createRequire as __esmCreateRequire } from "module"; import { fileURLToPath as __esmFileURLToPath } from "url"; import { dirname as __esmDirname } from "path"; const require = __esmCreateRequire(import.meta.url); const __filename = __esmFileURLToPath(import.meta.url); const __dirname = __esmDirname(__filename);',
```

Comment:
Could we add a comment why this is needed? I don't _think_ there is a need for this to be ESM, if we bundle as CJS can we drop this?